python/ray/tune/impl/tuner_internal.py

```diff
@@ -163,8 +164,58 @@ def fit(self) -> ResultGrid:
 
         return ResultGrid(analysis)
 
-    def _get_tune_run_arguments(self) -> Dict[str, Any]:
+    def _get_tune_run_arguments(self, trainable) -> Dict[str, Any]:
         """Get tune.run arguments common for both new and resumed runs."""
+        checkpoint_freq = self._run_config.checkpoint_config.checkpoint_frequency
+        checkpoint_at_end = self._run_config.checkpoint_config.checkpoint_at_end
+
+        if checkpoint_freq:
```

Comment:
should we ensure somewhere that checkpoint_freq is always >= 0?

Reply:
Added to `CheckpointConfig.__post_init__()`